edge/pkg/devicetwin/dtmanager/twin_test.go

```diff
@@ -0,0 +1,1534 @@
+	"github.com/kubeedge/kubeedge/edge/pkg/devicetwin/dttype"
+)
+
+var mockOrmer *beego.MockOrmer
+var mockQuerySeter *beego.MockQuerySeter
+
+// mocksInit is function to mock DBAccess
+func mocksInit(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockOrmer = beego.NewMockOrmer(mockCtrl)
+	mockQuerySeter = beego.NewMockQuerySeter(mockCtrl)
+	dbm.DBAccess = mockOrmer
+}
+
+// sendMsg is function to send message to receiverChannel and heartbeatChannel
+func (tw TwinWorker) sendMsg(msg *dttype.DTMessage, msgHeart string, actionType string, contentType interface{}) {
+	if tw.ReceiverChan != nil {
+		msg.Action = actionType
+		msg.Msg.Content = contentType
+		tw.ReceiverChan <- msg
+	}
+	if tw.HeartBeatChan != nil {
+		tw.HeartBeatChan <- msgHeart
+	}
+}
+
+// receiveMsg is function to receive message from the commChannel
+func receiveMsg(commChannel chan interface{}, message *dttype.DTMessage) {
+	msg, ok := <-commChannel
+	if !ok {
+		log.LOGGER.Errorf("No message received from communication channel")
+		return
+	}
+	*message = *msg.(*dttype.DTMessage)
+	return
+}
+
+// expectedFunc is function to declare an object of type TwinValue
+func expectedFunc() *dttype.TwinValue {
+	var expected dttype.TwinValue
+	value := "value"
+	valueMetaData := &dttype.ValueMetadata{Timestamp: time.Now().UnixNano() / 1e6}
+	expected.Value = &value
+	expected.Metadata = valueMetaData
+	return &expected
+}
+
+// actualFunc is function to declare an object of type TwinValue
+func actualFunc() *dttype.TwinValue {
+	var actual dttype.TwinValue
+	value := "value"
+	valueMetaData := &dttype.ValueMetadata{Timestamp: time.Now().UnixNano() / 1e6}
+	actual.Value = &value
+	actual.Metadata = valueMetaData
+	return &actual
+}
+
+// keyTwinUpdateFunc is function to declare an object of type DeviceTwinUpdate
+func keyTwinUpdateFunc() dttype.DeviceTwinUpdate {
+	var keyTwinUpdate dttype.DeviceTwinUpdate
+	twinKey := make(map[string]*dttype.MsgTwin)
+	twinKey["key1"] = &dttype.MsgTwin{
+		Expected: expectedFunc(),
+		Actual:   actualFunc(),
+		Metadata: &dttype.TypeMetadata{"nil"},
+	}
+	keyTwinUpdate.Twin = twinKey
+	keyTwinUpdate.BaseMessage = dttype.BaseMessage{EventID: "Event1"}
+	return keyTwinUpdate
+}
+
+// twinWorkerFunc is function to return an object of type TwinWorker
+func twinWorkerFunc(receiverChannel chan interface{}, confirmChannel chan interface{}, heartBeatChannel chan interface{}, context dtcontext.DTContext, group string) TwinWorker {
+	return TwinWorker{
+		Worker{
+			receiverChannel,
+			confirmChannel,
+			heartBeatChannel,
+			&context,
+		},
+		group,
+	}
+}
+
+// contextFunc is function to return an object of type DTContext
+func contextFunc() dtcontext.DTContext {
+	context := dtcontext.DTContext{
+		DeviceList:  &sync.Map{},
+		DeviceMutex: &sync.Map{},
+		Mutex:       &sync.Mutex{},
+	}
+	var testMutex sync.Mutex
+	context.DeviceMutex.Store("DeviceB", &testMutex)
```

Comment:
instead of hard-coding "DeviceB" , can we pass the device name as a parameter to the func ?